### PR TITLE
Fix timeout bug in Genesis “Time limited leashing attack” test

### PR DESCRIPTION
**Pull request by Niols**
See https://github.com/IntersectMBO/ouroboros-consensus/pull/1179 for context.

Note 1: I am currently running extensive tests to check that this fix looks good indeed. I wouldn't want to reintroduce an other extremely rare bug! I will mark this PR as ready once I am happy with the result, hopefully tomorrow :crossed_fingers: 

Note 2: There is a big more than just the fix, because I thought that some of my debugging work could be kept for posterity. Feel free to tell me to discard some bits and pieces, the commits are very independent for a reason.

### CODE REVIEW

File: ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Setup/GenChains.hs [hunk 1]
```diff
@@ -186,21 +180,16 @@ chainSyncTimeouts t f =
...
     idleTimeout :: Maybe DiffTime
-    idleTimeout = Just 3673 -- taken from Ouroboros.Consensus.Node.stdChainSyncTimeout
-    -- | The following timeout is derived from the average length of a streak of
-    -- empty slots. If the probability of the election of a leader is @f@ and
-    -- @Y@ is a probability, then a streak of empty slots will be shorter than
-    -- @log (1 - Y) / log (1 - f)@ with probability @Y@. Main net nodes pick a
-    -- random value for @Y@ between 99.9% and 99.999%. For our use case, we
-    -- choose the tightest bound of 99.9%.
+    -- | The default from 'Ouroboros.Consensus.Node.stdChainSyncTimeout' is
+    -- 3673s, which is virtally infinite, so let us make it actually infinite
```
Comment (jasagredo) [inline ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Setup/GenChains.hs:184]:
```suggestion
    -- 3673s, which is virtually infinite, so let us make it actually infinite
```

**Review by jasagredo — APPROVED** [review]
Looks good!

**Review by nfrisby — APPROVED** [review]
LGTM. Thanks for hunting this down so quickly.

I think the "extra" stuff seems useful, so I agree it's worthwhile to retain.

I made one suggestion for enriching one comment.

File: ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Setup/GenChains.hs [hunk 2]
```diff
@@ -186,21 +180,16 @@ chainSyncTimeouts t f =
...
-    -- choose the tightest bound of 99.9%.
+    -- | The default from 'Ouroboros.Consensus.Node.stdChainSyncTimeout' is
+    -- 3673s, which is virtually infinite, so let us make it actually infinite
+    -- for our test environment.
+    idleTimeout = Nothing
+    -- | The 'mustReplyTimeout' must be disabled in our context, because the
+    -- chains are finite, and therefore an honest peer can only serve it all,
+    -- then send 'MsgAwaitReply' (therefore entering 'StMustReply'), and then
+    -- stall forever, and it must not be killed for it.
     mustReplyTimeout :: Maybe DiffTime
```
[the remark this answers is not in the document]
Comment (Niols) [reply ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Setup/GenChains.hs:197]: What do you think about a36fd875dd1c7068cff4e4a39bcbdc11d9fdd1be?

[the remark this answers is not in the document]
Comment (nfrisby) [reply ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Setup/GenChains.hs:197]: Looks good, thanks.

**Comment by Niols** [comment]
Ran 2M tests, all green; previously, this would have had a very high probability to trigger the bug, so I have very strong confidence that this was the right fix. Will merge this.

**Pull request merged**